src/python/pants/engine/internals/parametrize.py

```diff
@@ -56,7 +56,15 @@ class _MergeBehaviour(Enum):
     merge_behaviour: _MergeBehaviour = dataclasses.field(compare=False)
 
     def __init__(self, *args: str, **kwargs: Any) -> None:
-        object.__setattr__(self, "args", args)
+        for arg in args:
```

Comment:
This is copied from the `to_parameters` method which "lazily" checks the arguments. Seems like we should be doing more eager error checks.